src/Feldspar/Compiler/Backend/C/CodeGeneration.hs

```diff
@@ -285,23 +285,14 @@ call fn args = fn <> parens (hsep $ punctuate comma args)
 --
 -- First parameter is whether the initialization context is somewhere
 -- inside a compound type.
-initialize :: Bool -> Type -> Doc
+initialize :: Type -> Doc
 -- Compound/Special types.
-initialize _     (1 :# Pointer{})  = equals <+> text "NULL"
-initialize _     ArrayType{}       = equals <+> text "NULL"
-initialize _     (NativeArray _ t) = equals <+> lbrace <+> initialize False t <+> rbrace
-initialize _     (StructType _ fs) = equals <+> lbrace <+> inits <+> rbrace
-  where inits = hsep $ punctuate comma $ map initField fs
-        initField (n, t) = char '.' <> text n <+> initialize True t
+initialize (1 :# Pointer{})  = equals <+> text "NULL"
+initialize ArrayType{}       = equals <+> text "NULL"
+initialize (NativeArray _ t) = equals <+> lbrace <+> initialize t <+> rbrace
```

Comment:
The new code looks better but initialize now looks partial to me, shouldn't this RHS also be text 0?

Reply:
Do you mean NativeArray? I was a bit unsure about that, so I left it as is. It is not going to be a very large initializer anyway; the problem this patch solves is very very large structs, and they are not affected by the RHS for NativeArray.

Reply:
Yes, but the output for NativeArray of Float is equals lbrace rbrace right now and I don't think that passes a C-compiler. (Even if it does, this code is meant to initialize the array, so the zero should be there.)

Reply:
I see your point. Will do.